arangod/Aql/ClusterNodes.cpp

```diff
@@ -423,20 +423,28 @@ CostEstimate DistributeNode::estimateCost() const {
 
   while (node) {
     switch (node->getType()) {
-      case ENUMERATE_COLLECTION:
-        return castTo<EnumerateCollectionNode const*>(node)->collection();
+      case ENUMERATE_COLLECTION: {
+        auto const* enumNode = castTo<EnumerateCollectionNode const*>(node);
+
+        if (!enumNode->isUsedAsSatellite() && enumNode->prototypeCollection() == nullptr) {
+          return enumNode->collection();
+        }
+        break;
+      }
       case INDEX:
         return castTo<IndexNode const*>(node)->collection();
       case TRAVERSAL:
       case SHORTEST_PATH:
       case K_SHORTEST_PATHS:
         return castTo<GraphNode const*>(node)->collection();
```

Comment:
Don't we need the same change (skipping satellites) not just for `ENUMERATE_COLLECTION`, but also for `INDEX` and the GraphNodes?
Also, you can combine the `ENUMERATE_COLLECTION` and `INDEX` cases by casting them to `CollectionAccessingNode`s.